crates/oxc_linter/src/rules/eslint/no_await_in_loop.rs

```diff
@@ -0,0 +1,233 @@
+use oxc_ast::{
+    ast::{Expression, Statement},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-await-in-loop): Unexpected `await` inside a loop.")]
+#[diagnostic(severity(warning))]
+struct NoAwaitInLoopDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoAwaitInLoop;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// This rule disallows the use of await within loop bodies. (for, for-in, for-of, while, do-while).
+    ///
+    /// ### Why is this bad?
+    ///
+    /// It potentially indicates that the async operations are not being effectively parallelized.
+    /// Instead, they are being run in series, which can lead to poorer performance.
+    ///
+    /// ### Example
+    /// Bad:
+    /// ```javascript
+    /// for (const user of users) {
+    ///   const userRecord = await getUserRecord(user);
+    /// }
+    /// ```
+    ///
+    /// Good:
+    /// ```javascript
+    /// await Promise.all(users.map(user => getUserRecord(user)));
+    /// ```
+    NoAwaitInLoop,
+    correctness
```

Comment:
Is [`perf`](https://github.com/oxc-project/oxc/blob/32504cadb745c4f81be89fa74fb317822daee1ed/crates/oxc_linter/src/rule.rs#L42-L43) much better?